### ISSUE TITLE
tracking on train\val loss function in real time

### ISSUE THREAD

**Issue by Liranbz** [allegroai/trains#179]
Hi,
I`m trying to visualize the train\val loss function in real time. currently I print every epoch the metrics. here is my code:

`
print(f'Epoch {e + 0:03}:'
              f' | Train Loss: {train_epoch_loss / len(train_loader):.5f} |'
              f' Val Loss: {val_epoch_loss / len(val_loader):.5f} |'
              f' Train Acc: {train_epoch_acc / len(train_loader):.3f}|'
              f' Val Acc: {val_epoch_acc / len(val_loader):.3f} |'
              f' Train F_score: {train_epoch_f_score / len(train_loader):.3f}|'
              f' Val F_score: {val_epoch_f_score / len(val_loader):.3f}')
`
I got the plots of these metrics only In the end of the training. Is there a way to visualize it in real time\ each epoch with trains.ai?
Thank you!

**Comment by jkhenning** [comment]
Hi @Liranbz ,

How do you log your metrics? If you're using Tensorboard or Matplotlib, or when you call the Trains `Logger` explicitly, the metrics are reported in real time - which one do you use?

**Comment by Liranbz** [comment]
Hi @jkhenning ,
Currently I get the plots only in the end of the training. At each epoch I save the items to arrays:
```
loss_stats['train'].append(train_epoch_loss / len(train_loader))
loss_stats['val'].append(val_epoch_loss / len(val_loader))
accuracy_stats['train'].append(train_epoch_acc / len(train_loader))
 accuracy_stats['val'].append(val_epoch_acc / len(val_loader))
f_score_stats['train'].append(train_epoch_f_score / len(train_loader))
f_score_stats['val'].append(val_epoch_f_score / len(val_loader))
```
and got the plots with matplotlib plots..
My question is how to do it in real time with trains.ai? (If there is a way to do it..)
Thank you

**Comment by jkhenning** [comment]
Hi @Liranbz ,

Whatever you plot with `matplotlib` is reported in real time - since you only plot it at the end of training, real-time is the end 😄 

If you want to report during training, you can easily do that using the Trains `Logger` object.

At each epoch, simply use `Logger.report_scalar()` to report whatever value you'd like. The call requires an `iteration` value, which in your case can simply be a running number (incremented every epoch), so for example:

```python
from trains import Task
task = Task.init(project_name="my project", task_name="my task")

# initialize an iteration counter when starting
iteration = 0

...

# At each epoch, report using Trains Logger
logger = task.get_logger()
logger.report_scalar("loss_stats", "train", iteration=iteration, value=train_epoch_loss / len(train_loader))
logger.report_scalar("loss_stats", "val", iteration=iteration, value=val_epoch_loss / len(val_loader))
logger.report_scalar("accuracy_stats", "train", iteration=iteration, value=train_epoch_acc / len(train_loader))
logger.report_scalar("accuracy_stats", "val", iteration=iteration, value=val_epoch_acc / len(val_loader))
logger.report_scalar("f_score_stats", "train", iteration=iteration, value=train_epoch_f_score / len(train_loader))
logger.report_scalar("f_score_stats", "val", iteration=iteration, value=val_epoch_f_score / len(val_loader))
```

You can see a very similar example [here](https://github.com/allegroai/trains/blob/master/examples/reporting/scalar_reporting.py), and more reporting examples [here](https://github.com/allegroai/trains/tree/master/examples/reporting)

**Comment by Liranbz** [comment]
Hi @jkhenning,
I know :) that's the reason for my question. Thank you for your response, now it`s working well! :)
